todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt

```diff
@@ -15,109 +15,64 @@
-                val task = Task(title, description, itemId).apply {
-                    isCompleted = getInt(getColumnIndexOrThrow(COLUMN_NAME_COMPLETED)) == 1
+        appExecutors.diskIO.execute {
+            val tasks = tasksDao.getTasks()
+            appExecutors.mainThread.execute {
+                if (tasks.isEmpty()) {
+                    // This will be called if the table is new or just empty.
+                    callback.onDataNotAvailable()
+                } else {
+                    callback.onTasksLoaded(tasks)
                 }
-                tasks.add(task)
             }
-            if (tasks.isNotEmpty()) {
-                callback.onTasksLoaded(tasks)
-            } else {
-                // This will be called if the table is new or just empty.
-                callback.onDataNotAvailable()
-            }
-            close()
         }
-        db.close()
     }
 
     /**
      * Note: [GetTaskCallback.onDataNotAvailable] is fired if the [Task] isn't
      * found.
      */
     override fun getTask(taskId: String, callback: TasksDataSource.GetTaskCallback) {
-        val db = dbHelper.readableDatabase
-
-        val projection = arrayOf(COLUMN_NAME_ENTRY_ID, COLUMN_NAME_TITLE,
-                COLUMN_NAME_DESCRIPTION, COLUMN_NAME_COMPLETED)
-
-        val cursor = db.query(
-                TABLE_NAME, projection, "$COLUMN_NAME_ENTRY_ID LIKE ?", arrayOf(taskId), null,
-                null, null)
-
-        with(cursor) {
-            if (moveToFirst()) {
-                val itemId = getString(getColumnIndexOrThrow(COLUMN_NAME_ENTRY_ID))
-                val title = getString(getColumnIndexOrThrow(COLUMN_NAME_TITLE))
-                val description = getString(getColumnIndexOrThrow(COLUMN_NAME_DESCRIPTION))
-                val task = Task(title, description, itemId).apply {
-                    isCompleted = getInt(getColumnIndexOrThrow(COLUMN_NAME_COMPLETED)) == 1
+        appExecutors.diskIO.execute {
+            val task = tasksDao.getTaskById(taskId)
+            appExecutors.mainThread.execute {
+                if (task != null) {
+                    callback.onTaskLoaded(task)
+                } else {
+                    callback.onDataNotAvailable()
                 }
-                callback.onTaskLoaded(task)
-            } else {
-                callback.onDataNotAvailable()
             }
-            close()
         }
-        db.close()
     }
 
     override fun saveTask(task: Task) {
-        val values = ContentValues().apply {
-            put(COLUMN_NAME_ENTRY_ID, task.id)
-            put(COLUMN_NAME_TITLE, task.title)
-            put(COLUMN_NAME_DESCRIPTION, task.description)
-            put(COLUMN_NAME_COMPLETED, task.isCompleted)
-        }
-        with(dbHelper.writableDatabase) {
-            insert(TABLE_NAME, null, values)
-            close()
+        appExecutors.diskIO.execute {
```

Comment:
join lines (and throughout for these statements)

Reply:
Done (cherry-picked dev-todo-mvp-kotlin modifications).